### PR TITLE
Cultist teleport rune improvements

### PR DESCRIPTION
**Pull request by AlaunusLux**
I noticed during a game that the teleportation talisman was inconsistent. I can't remember if I was able to make it work during that game, but during testing I wasn't able to get them to work at all. 

This adds a `network` variable to the talisman, which is only set if the infusing rune is a teleportation rune.

This annoyingly will now delete the talisman properly upon use. I was tempted to fix the teleportation but not the proc call ;)

There is also at least one phantom teleportation rune, which has a null parent, so would 'use' the rune and set its cooldown without being able to teleport the user, so there's a check for that now. I believe the phantom/orphan runes are created when a talisman is created. Will look into a fix, but it doesn't affect rune count, just makes the for loop work a bit harder.

This also stops the rune and talisman from sending a "sizzled with no result" message even when it works.

Also will show which network the talisman is keyed to on examination. Added a missing space in the rune examine text.

### CODE REVIEW

**Comment by AlaunusLux** [comment]
!bugfix !review

**Comment by AlaunusLux** [comment]
!bugfix

File: code/game/gamemodes/cult/items/talisman.dm [hunk 1]
```diff
@@ -17,7 +18,10 @@
 /obj/item/paper/talisman/examine(mob/user)
 	. = ..()
 	if(iscultist(user) && rune)
-		to_chat(user, "The spell inscription reads: <span class='cult'><b><i>[rune.name]</i></b></span>.")
+		var/network_text = ""
+		if(network)
+			network_text = " This spell's network tag reads: <span class='cult'><b><i>[network]</i></b></span>."
+		to_chat(user, "The spell inscription reads: <span class='cult'><b><i>[rune.name]</i></b></span>.[network_text]")
```
Comment (FluffyGhoster) [inline code/game/gamemodes/cult/items/talisman.dm:24]: Use SPAN_CULT here

File: code/game/gamemodes/cult/runes/teleport.dm [hunk 2]
```diff
@@ -16,7 +17,7 @@
 /datum/rune/teleport/get_cultist_fluff_text()
 	. = ..()
 	if(network)
-		. += "This rune's network tag reads: <span class='cult'><b><i>[network]</i></b></span>."
+		. += " This rune's network tag reads: <span class='cult'><b><i>[network]</i></b></span>."
```
Comment (FluffyGhoster) [inline code/game/gamemodes/cult/runes/teleport.dm:20]: SPAN_CULT

File: code/game/gamemodes/cult/runes/teleport.dm [hunk 3]
```diff
@@ -34,13 +35,15 @@
...
+	var/obj/item/paper/talisman/teleport_talisman = A
+	if(teleport_talisman.uses < 1)
+		qdel(A)
 
 /datum/rune/teleport/do_rune_action(mob/living/user, atom/movable/A)
-	teleport(user, A)
+	teleport(user, A, TRUE)
 
-/datum/rune/teleport/proc/teleport(mob/living/user, atom/movable/A)
+/datum/rune/teleport/proc/teleport(mob/living/user, atom/movable/A, is_rune)
```
Comment (FluffyGhoster) [inline code/game/gamemodes/cult/runes/teleport.dm:46]:
```suggestion
/datum/rune/teleport/proc/teleport(mob/living/user, atom/movable/A, is_rune = FALSE)
```

File: code/game/gamemodes/cult/items/talisman.dm (hunk 1 above)
Comment (AlaunusLux) [reply code/game/gamemodes/cult/items/talisman.dm:24]: I'm glad that exists, looking at the repetitive markup was hurting me.

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**